### PR TITLE
[RTL] Fixed issues with numeric and time formatting in RTL mode.

### PR DESCRIPTION
**Pull request by Siemienik**
### Context
Based on https://github.com/handsontable/handsontable/pull/8893 ([diff](https://github.com/handsontable/handsontable/compare/feature/issue-8795...feature/issue-8791))

Fixed issues with numeric and time formatting in RTL mode. Setting `dir=ltr` forces to display the correct value as formatted in settings. For another format, it should be achieved by format on the settings level.

Resolves: #8805 #8791

### How has this been tested?

I used configurations from the issues. Then I checked for both LTR and RTL mode. Additionally, I update the test suits to cover these changes.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. Resolves: #8805 
2. Resolves: #8791
3. Part of #8760 

### Affected project(s):
- [x] `handsontable`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)

### CODE REVIEW

**Review by wszymanski — APPROVED** [review]

**Comment by wszymanski** [comment]
I'm sorry, I've accepted this PR by mistake. I'll review it soon.

**Review by wszymanski — DISMISSED** [review]
I'm sorry, I've accepted this PR by mistake. I'll review it soon.

**Review by budnix — CHANGES_REQUESTED** [review]
I left my suggestions in the comments.

File: handsontable/src/renderers/textRenderer/textRenderer.js [hunk 1]
```diff
@@ -44,6 +44,9 @@ export function textRenderer(instance, TD, row, col, prop, value, cellProperties
     // this is faster than innerHTML. See: https://github.com/handsontable/handsontable/wiki/JavaScript-&-DOM-performance-tips
     fastInnerText(TD, escaped);
   }
+  if (cellProperties.type === 'time') {
```
Comment (budnix) [inline handsontable/src/renderers/textRenderer/textRenderer.js:47]: The same case is mentioned here https://github.com/handsontable/handsontable/pull/8906#discussion_r763077563

File: handsontable/src/renderers/numericRenderer/numericRenderer.js [hunk 2]
```diff
@@ -49,6 +49,8 @@ export function numericRenderer(instance, TD, row, col, prop, value, cellPropert
     }
 
     cellProperties.className = classArr.join(' ');
+
+    TD.dir = 'ltr';
```
Comment (budnix) [inline handsontable/src/renderers/numericRenderer/numericRenderer.js:53]: The TD elements that are injected by Walkontable are reused during the HoT usage. To make sure that the TD attributes do not go over to another cell (after CRUD operations, hiding columns/rows, etc) the TD attribute has to be cleared.

May I suggest updating [the line of code](https://github.com/handsontable/handsontable/blob/feature/issue-8791/handsontable/src/3rdparty/walkontable/src/renderer/cells.js#L92) and add `TD.removeAttribute('dir');`?

Comment (budnix) [reply handsontable/src/renderers/numericRenderer/numericRenderer.js:53]: Resolved within 92a351e

**Comment by budnix** [comment]
I finished working on this PR. I implemented all of the ideas and suggestions that I had while reviewing the code. If @Siemienik approves the changes the PR can be merged. Things that I pushed:
 * Removed `dir` attribute from the cell element within Walkontable's cells renderer (92a351ef79566a6fd2aa5bcfbd0cb6ea7a881958) and cover the changes with the test (361699ffd8fc884efbba236a4f3be285ae80e1f5);
 * Following the separation of concern I removed the `time` logic that was originally placed within the TextEditor and text renderer I create new TimeEditor and time renderer with a proper RTL changes (ce7a709ca78d6cf001148f4eb9bfb509d3819e8a, 12def75447482ca6e21f87532930f95d27935d75).
 * Added the RTL tests for all editors and renderers to make sure that the "dir" attribute change within other editors and renderers will be caught.

**Comment by Siemienik** [comment]
Thank you @budnix for your improvements, It looks good to me 🥇. IMHO, we may merge this.

**Review by budnix — APPROVED** [review]

**Pull request merged**